Tests/StructuredQueriesTests/SelectTests.swift

```diff
@@ -1402,6 +1402,20 @@ extension SnapshotTests {
         }
       }
     }
+
+    @Test func singleJoinChaining() {
+      let base = Reminder.group(by: \.id).join(ReminderTag.all) { $0.id.eq($1.reminderID) }
```

Comment:
it doesn't test on the actual bug.

it is always working, without patch also working:
`Reminder.group(by: \.id).join(ReminderTag.all) { $0.id.eq($1.reminderID) }`

tricky is only not working at this case:
`Reminder.group(by: \.id).withReminderTagJoin`

Reply:
While this line compiles fine, the lines below it triggered what I think were equivalent compiler errors without the overloads defined in this PR. The issue doesn't seem to be related to the static dynamic member lookup.